x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -292,7 +292,11 @@ private static boolean isBasic(OperationMode mode) {
     private final boolean isSecurityEnabled;
     private final boolean isSecurityExplicitlyEnabled;
 
-    private Status status = new Status(OperationMode.TRIAL, true);
+    // Since Status is the only field that can be updated, we do not need to synchronize access to
+    // XPackLicenseState. However, if status is read multiple times in a method, a local variable should be
+    // be used to avoid update races. If additional volatile state is added to XPackLicenseState, we may need
+    // to reintroduce synchronization.
+    private volatile Status status = new Status(OperationMode.TRIAL, true);
```

Comment:
I wonder if because of the frequency of the pattern of reading status multiple times in a single method, a safer approach would be the use of a read/write lock. This way, future new method that read the status multiple times would be less likely to be coded incorrectly, since the pattern around reads elsewhere would clearly stand out as first requiring grabbing the read lock.